watertap/unit_models/zero_order/pump_variable_zo.py

```diff
@@ -0,0 +1,133 @@
+    """
+
+    CONFIG = ZeroOrderBaseData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        self._tech_type = "pump_variable"
+
+        build_pt(self)
+
+        self.lift_height = Var(units=pyunits.m, doc="Lift height for pump")
+
+        self.eta_pump = Var(units=pyunits.dimensionless, doc="Efficiency of pump")
+
+        self.eta_motor = Var(units=pyunits.dimensionless, doc="Efficiency of motor")
+
+        self.flow_bep = Var(
+            units=pyunits.m**3 / pyunits.s, doc="Best efficiency point flowrate"
+        )
+
+        self.flow_ratio = Var(
+            self.flowsheet().config.time,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Ratio between instantaneous flowrate and best efficiency point flowrate",
+        )
+
+        self.eta_ratio = Var(
+            self.flowsheet().config.time,
+            units=pyunits.dimensionless,
+            bounds=(0, 1),
+            doc="Ratio between the true efficiency and the best efficiency due to variable operation",
+        )
+        self.electricity = Var(
+            self.flowsheet().config.time,
+            units=pyunits.kW,
+            bounds=(0, None),
+            doc="Electricity for low pressure pump",
+        )
+
+        self.applied_pressure = Var(
+            self.flowsheet().config.time,
+            units=pyunits.bar,
+            bounds=(0, None),
+            doc="Applied pressure",
+        )
+
+        self._fixed_perf_vars.append(self.lift_height)
+        self._fixed_perf_vars.append(self.eta_pump)
+        self._fixed_perf_vars.append(self.eta_motor)
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for pump applied pressure"
+        )
+        def applied_pressure_constraint(b, t):
+            return b.applied_pressure[t] == pyunits.convert(
+                b.lift_height
+                * b.properties[t].dens_mass
+                * Constants.acceleration_gravity,
+                to_units=pyunits.bar,
+            )
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for variable pump flowrate ratio"
+        )
+        def flow_ratio_constraint(b, t):
+            return b.flow_ratio[t] == b.properties[t].flow_vol / pyunits.convert(
+                b.flow_bep, to_units=pyunits.m**3 / pyunits.s
+            )
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for variable pump efficiency"
+        )
+        def eta_ratio_constraint(b, t):
+            # if the flow is too far from the design point, the efficiency is set to 40%
+            if b.flow_ratio[t] < 0.6 or b.flow_ratio[t] > 1.4:
+                return b.eta_ratio[t] == 0.4
+            # otherwise, use the correlation
+            else:
+                return b.eta_ratio[t] == (
+                    -0.995 * b.flow_ratio[t] ** 2 + 1.977 * b.flow_ratio[t] + 0.025
+                )
```

Comment:
I think you can start adding tests. I expect that this might not work out, but let's see what happens after testing.